Tests/SentryTests/Performance/SentryTracerTests.swift

```diff
@@ -136,11 +150,23 @@ class SentryTracerTests: XCTestCase {
             XCTAssertEqual(tracerTimestamp.timeIntervalSince1970, span["timestamp"] as? TimeInterval)
         }
     }
+
+    func testDeadlineTimer_FinishesTransactionAndChildren() {
```

Comment:
`m`: Please add a test that tests the proper behavior when having a transaction with multiple unfinished children and a couple of finished children.

Also, add a test that validates that the timeout is canceled correctly when finishing the transaction.

Reply:
Both done 👍